### DISCUSSION TITLE
HuggingFace Integration/Tips

### DISCUSSION THREAD

**Discussion by piyushnags** [ContinualAI/avalanche#1418] [Feature Request] [answered]
First of all, I'd like to say that I really like the usability of the Avalanche framework, especially how flexible it is. After looking through the examples in the Github repos, I saw a few examples about how to integrate Hugging Face models into the framework. At some point, I would like to see a concrete integration of HuggingFace models into the Avalanche framework including the Computer Vision models.

In the short term, can we get a post about the things we need to consider while integrating HF models? For example, when I tried to integrate an HF model with the EWC plugin, I had to rewrite the plugin so that the loss was computed correctly even though I updated the strategy I was using (SupervisedTemplate). In this case, the code crashed, so it was easy enough to figure out what needed to be done. However, I worry that there might be subtle bugs that could go undetected and it would be helpful to have some guidelines to refer to.

Thanks

**Comment by vlomonaco** [comment]
@AndreaCossu do you thave some additional insights on this?

**Comment by AndreaCossu** [comment, marked as answer]
Hi @piyushnags and thanks for reaching out. You are right, at the moment there is no official support for HF API and we are handling it on a case-by-case basis. We can definitely think about publishing a tutorial, similar to other ones we currently have in our website. In the meantime, if you encounter specific problems feel free to open an issue and we can try to help you out.
